Telemetry/TelemetryStorage.swift

```diff
@@ -170,6 +170,63 @@ public class TelemetryStorage {
         }
         return nil
     }
+}
 
+// Event array storage handling
+extension TelemetryStorage {
 
+    func eventArrayFile(forPingType pingType: String) -> URL? {
+        do {
+            let url = try FileManager.default.url(for: configuration.dataDirectory, in: .userDomainMask, appropriateFor: nil, create: true).appendingPathComponent("eventArray-\(name)-\(pingType).json")
+            return url
+        } catch {
+            print("\(#function) \(error)")
+            return nil
+        }
+    }
+
+    func deleteEventArrayFile(forPingType pingType: String) {
+        guard let url = eventArrayFile(forPingType: pingType), FileManager.default.fileExists(atPath: url.path) else {
+            return
+        }
+
+        do {
+            try FileManager.default.removeItem(at: url)
+        } catch {
+            print("\(#function) \(error)")
+        }
+    }
+
+    // Appends JSON array, the file is a JSON snippet like "[event1],[event2]".
+    // To read this file: read to a string, wrap the string in '[ ]' to complete the JSON,
+    // then pass to JSON parser.
+    func appendEvent(data: Data, forPingType pingType: String) {
```

Comment:
As mentioned earlier, it might be clearer to pass in a `TelemetryEvent` here and return a `Bool` indicating if the event was actually written.